clap_generate/src/man.rs

```diff
@@ -0,0 +1,287 @@
+        //     None => out.push_str(r#" """#),
+        // }
+
+        buf.push_str(r#" """#);
+
+        match self.app.get_version() {
+            Some(version) => write(
+                buf,
+                format_args!(r#" "{} {}""#, self.app.get_name(), version),
+            )
+            .unwrap(),
+            None => buf.push_str(r#" """#),
+        }
+
+        match self.manual.as_ref() {
+            Some(manual) => write(buf, format_args!(r#" "{}""#, manual)).unwrap(),
+            None => buf.push_str(r#" """#),
+        }
+
+        buf.push('\n');
+    }
+
+    fn render_about(&self, buf: &mut String) {
+        buf.push_str(".SH \"NAME\"\n");
+        buf.push_str(&bold(self.app.get_name()));
+
+        if let Some(about) = self.app.get_about() {
+            buf.push_str(r" \- ");
+            buf.push_str(&escape(about));
+        }
+
+        buf.push('\n');
+    }
+
+    fn render_description(&self, buf: &mut String) {
+        if let Some(about) = self.app.get_long_about() {
+            buf.push_str(".SH \"DESCRIPTION\"\n");
+            buf.push_str(about);
+            buf.push('\n');
+        }
+    }
+
+    fn render_synopsis(&self, buf: &mut String) {
+        buf.push_str(".SH \"SYNOPSIS\"\n");
+        buf.push_str(&italic(self.app.get_name()));
+        buf.push(' ');
+
+        for opt in self.app.get_arguments() {
+            match (opt.get_short(), opt.get_long()) {
+                (Some(short), Some(long)) => {
+                    write(buf, format_args!("[-{}|--{}] ", short, long)).unwrap()
+                }
+                (Some(short), None) => write(buf, format_args!("[-{}] ", short)).unwrap(),
+                (None, Some(long)) => write(buf, format_args!("[--{}] ", long)).unwrap(),
+                (None, None) => (),
+            }
+        }
+
+        for arg in self.app.get_positionals() {
+            write(buf, format_args!("[{}]", arg.get_name())).unwrap();
+        }
+
+        buf.push('\n');
+    }
+
+    // TODO: handle positional arguments in a different way?
+    fn render_options(&self, buf: &mut String) {
+        let items: Vec<_> = self
+            .app
+            .get_arguments()
+            .filter(|i| !i.is_set(ArgSettings::Hidden))
+            .collect();
+        if items.is_empty() {
+            return;
+        }
+
+        buf.push_str(".SH \"OPTIONS\"\n");
+        for opt in items.iter().filter(|a| !a.is_positional()) {
+            let args = match (opt.get_short(), opt.get_long()) {
+                (Some(short), Some(long)) => format!("-{}, --{}", short, long),
+                (Some(short), None) => format!("-{}", short),
+                (None, Some(long)) => format!("--{}", long),
+                (None, None) => "".to_string(),
+            };
+
+            buf.push_str(&bold(&args));
+
+            if let Some(value) = &opt.get_value_names() {
+                buf.push(' ');
+                buf.push_str(&italic(&value.join(" ")));
+            }
```

Comment:
It seems to work when generating it.

Reply:
Can you verify handling of value names with
- `Arg::new("foo").long("foo").takes_value(true).min_values(3).max_values(3)
- `Arg::new("foo")..takes_value(true).min_values(3).max_values(3)
- `Arg::new("foo")..takes_value(true).min_values(3).max_values(3).value_names(["common"])
- `Arg::new("foo")..takes_value(true).min_values(3).max_values(3).value_names(["one", "two", "three"])

Reply:
I had to rename the arguments since they had the same names, but this is the output with those values. 

```man
       -c, --config FILE
           Sets a custom config file

           Some more text about how to set a custom config file

       --foo

       [bar]

       [baz]

       [foobar]
```

Reply:
`Arg::new("foo").long("foo").takes_value(true).min_values(3).max_values(3)`
should lead to `--foo <foo> <foo> <foo>

Similar for `Arg::new("foo")..takes_value(true).min_values(3).max_values(3)` and `Arg::new("foo")..takes_value(true).min_values(3).max_values(3).value_names(["common"])`

I didn't see an example for `Arg::new("foo")..takes_value(true).min_values(3).max_values(3).value_names(["one", "two", "three"])` but it should lead to `<one> <two> <three>`